File: inject-java-test/src/test/groovy/io/micronaut/inject/visitor/beans/TestEntity.java
package io.micronaut.inject.visitor.beans;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.Id;
import javax.persistence.Version;
import javax.validation.constraints.Size;

@Entity
public class TestEntity {
    @Id
    @GeneratedValue
    private Long id;
    @Version
    private Long version;
    private String name;
    @Size(max=100)
    private int age;


    public String getName() {
        return this.name;
    }
    public void setName(String n) {
        this.name = n;
    }
    public int getAge() {
        return age;
    }
    public void setAge(int age) {
        this.age = age;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public Long getId() {
        return this.id;
    }

    public void setVersion(Long version) {
        this.version = version;
    }

    public Long getVersion() {
        return this.version;
    }
}

File: inject-kotlin/src/test/kotlin/io/micronaut/kotlin/processing/elementapi/TestEntity.kt
package io.micronaut.kotlin.processing.elementapi

import jakarta.validation.constraints.*
import javax.persistence.*

@Entity
class TestEntity(
    @Column(name="test_name") var name: String,
    @Size(max=100) var age: Int,
    primitiveArray: Array<Int>) {

    @Id
    @GeneratedValue
    var id: Long? = null

    @Version
    var version: Long? = null

    private var primitiveArray: Array<Int>? = null

    private var v: Long? = null

    @Version
    fun getAnotherVersion(): Long? {
        return v;
    }

    fun setAnotherVersion(v: Long) {
        this.v = v
    }
}
